src/server.c

```diff
@@ -3174,6 +3256,41 @@ void homekit_server_task(void *args) {
         INFO("Using existing accessory ID: %s", server->accessory_id);
     }
 
+    if (!server->config->setupId) {
+
+      char *setupIdentifier = malloc(5);
+
+        for (int i=0; i<4; i++) {
+            setupIdentifier[i] = base36Table[(hwrand() % 36)];
+        }
+
+        setupIdentifier[4] = 0;
+        server->config->setupId = setupIdentifier;
+
+        DEBUG("Using random setup identifier: %s", setupIdentifier);
+    }
+
+    if (!server->config->password) {
```

Comment:
The reason why this was left inside Pair Setup code instead of init code is that it generates random password ON EVERY NEW pairing attempt. It's kind of new random password every time you start pairing.

Reply:
In case of qr-code setup that seems to be impossible. The pairing endpoint gets called after the code was scanned. 

A solution could be to generate the pin at initialisation if the setup code is set or the URI callback is set. If not, then the pin could be generated at the pairing setup.

Reply:
The specification says:

>4.3.1 Displaying Setup Codes
>If an accessory can display a random Setup Code, it must conform to the following rules:
> ● The accessory must generate a new Setup Code each time it is needed.
> ...
>If an accessory cannot display a random Setup Code for any reason then it must conform to the following rules:
>● A random Setup Code must be generated for each individual accessory.

Based on this, the presence of the callbacks should determine the case. But it may be more transparent to select a pairing mode in the configuration ...

Reply:
Yes, that was the case: the presence of callback means that accessory should be able to communicate password to user. Although, if static password is present, then that one should be used instead of a random password. It is incorrect to not have a static password and callback at the same time.

Now, having all this your PR breaks support for "must generate a new Setup Code each time it is needed"

Reply:
The use case for a dynamic generated setupURI is the same as for dynamic generated passwords. Devices with a display to show pairing information.

But most likely a pincode or pairing information is supplied on the housing of the device